### ISSUE TITLE
Component broken in 0.94

### ISSUE THREAD

**Issue by Nick-Adams-AU** [Swamp-Ig/izone_custom_component#6]
I'm seeing some errors when the component is loaded in the recently released HA 0.94. I'm using the latest master branch of the component. I appreciate that this HA version has only just been released :)

`2019-06-07 10:26:45 ERROR (MainThread) [homeassistant.config_entries] Error importing integration izone to set up izone config entry: No module named 'custom_components.izone.config_flow'
2019-06-07 10:26:45 ERROR (MainThread) [homeassistant.config_entries] Error occurred loading config flow for integration izone: No module named 'custom_components.izone.config_flow'`

**Comment by Swamp-Ig** [comment]
I haven't updated to this version yet. Will take a look when I have time.

I should probably submit it to get included as a non-custom component really.

**Comment by Zac300** [comment]
Good morning, 
I too am seeing getting the exact same error once upgrading to 0.94.1. Happy to assist with testing as needed.

**Comment by Swamp-Ig** [comment]
I'm working on it :grin:

I am trying to make the testing a bit more robust, I have already identified a couple of issues that would show up with flaky network connections. I might well get you both to run a script that generates output that can be put into the test cases.

**Comment by Nick-Adams-AU** [comment]
Happy to help with testing! I've actually got two systems on the same network so that should help with some additional test cases.

**Comment by Swamp-Ig** [comment]
@Nick-Adams-AU  Oh really? And that actually works well? I did write for that but I didn't think it would actually happen!

**Comment by Nick-Adams-AU** [comment]
It seems to work well. 

The only issue I have is that they both display "0" as the temperature in the widget as I have no temperature probes on any zones. I'd much prefer to be able to supply the temperature from another sensor.

**Comment by teaching-innovation** [comment]
Hi Nick, I also had the issue with displaying "0" as the Temp and would prefer to use a custom Temp Sensor. I have the issue in this thread from most recent update so cannot currently see the climate component, however... I went into the iZone controller settings the other night and noticed that if changed the iZone temp control from "Return Air" to "Master" which I think is a temp sensor built into the touch screen.  When I looked at the JSON data by http://IP-ADDRESS/SystemSettings system temperature was no longer 0.
When I select Return Air the temp is displayed on the iZone touch panel but comes up as 0 in the system settings JSON. So iZone does get the temp from the AC unit but doesn't present it in the system settings. I did email iZone about the issue but didn't hear back.

**Comment by Swamp-Ig** [comment]
Yeh I've fixed the 0 thing already, if the temp is actually zero it's going to display like the temp is unknown, but meh. 

Should get the integration into an upcoming home assistant release :grin:

**Comment by Nick-Adams-AU** [comment]
Is it possible to feed in a custom sensor value instead of the system value? My return air or master temp are both irrelevant for me due to the configuration of the system.

**Comment by Swamp-Ig** [comment]
Not really, that would be outside the scope of a single integration. You'd use something like a template device to do that.

**Comment by Swamp-Ig** [comment]
OK good news everyone, I've just done a PR to get iZone integrated into HA directly, rather than as a custom component.

Feel free to watch the PR:  home-assistant/home-assistant#24550

**Comment by Nick-Adams-AU** [comment]
Great work! Thanks for your help. Much appreciated.

**Comment by Zac300** [comment]
Awesome, thanks @Swamp-Ig I look forward to getting this fix in a later release of home assistant

**Comment by Swamp-Ig** [comment]
If you want to you could probably download the component from the PR and install as a custom component in the mean time, it should work ok. Just put:

```
izone:
```

In the config file if it isn't already

**Comment by AtticusG3** [comment]
I really wish I knew how to grab the files from the PR. I'm setting up HA for a friend with an iZone aircon controller and If this integration works well I'll likely get one for my own HVAC.

Thank you for putting in the effort to make thi component.

**Comment by Zac300** [comment]
@Swamp-Ig thanks, i've grabbed the files from your PR and can confirm they work with 0.94.4. Can't wait for this to be merged 👍

**Comment by AtticusG3** [comment]
I'm still really looking forward to being able to use this component in the near future. I have an izone controller sitting on the wall acting dumb right now waiting. :-D

**Comment by Swamp-Ig** [comment]
Yep, I only get to work on it one day a week. Getting close though.
